From 006e73876fe1d00e040d3d149ed896d832dfa16c Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Mon, 4 Aug 2014 19:00:44 -0400
Subject: [fs] btrfs: fix max_inline mount option

Message-id: <1407179109-13714-158-git-send-email-sandeen@redhat.com>
Patchwork-id: 87103
O-Subject: [RHEL7.1 PATCH 157/422] Btrfs: fix max_inline mount option
Bugzilla: 1071179
RH-Acked-by: Zach Brown <zab@redhat.com>

From: Mitch Harder <mitch.harder@sabayonlinux.org>

commit feb5f96589302b39a2b10fc210db7c47a73e4168
Author: Mitch Harder <mitch.harder@sabayonlinux.org>
Date:   Thu Feb 13 09:13:16 2014 -0600

    Btrfs: fix max_inline mount option

    Currently, the only mount option for max_inline that has any effect is
    max_inline=0.  Any other value that is supplied to max_inline will be
    adjusted to a minimum of 4k.  Since max_inline has an effective maximum
    of ~3900 bytes due to page size limitations, the current behaviour
    only has meaning for max_inline=0.

    This patch will allow the the max_inline mount option to accept non-zero
    values as indicated in the documentation.

    Signed-off-by: Mitch Harder <mitch.harder@sabayonlinux.org>
    Signed-off-by: Chris Mason <clm@fb.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/btrfs/super.c b/fs/btrfs/super.c
index 97cc241..e73c80e 100644
--- a/fs/btrfs/super.c
+++ b/fs/btrfs/super.c
@@ -566,7 +566,7 @@ int btrfs_parse_options(struct btrfs_root *root, char *options)
     kfree(num);
 
     if (info->max_inline) {
-     info->max_inline = max_t(u64,
+     info->max_inline = min_t(u64,
       info->max_inline,
       root->sectorsize);
     }
-- 
1.7.1